# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/TransformFeedback.h
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/libANGLE/TransformFeedback.h
--- a/gfx/angle/checkout/src/libANGLE/TransformFeedback.h	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/TransformFeedback.h	Sat Sep 07 13:38:36 2019 +0000
@@ -38,6 +38,8 @@
     const OffsetBindingPointer<Buffer> &getIndexedBuffer(size_t idx) const;
     const std::vector<OffsetBindingPointer<Buffer>> &getIndexedBuffers() const;
     const Program *getBoundProgram() const { return mProgram; }
+    GLsizeiptr getVerticesDrawn() const { return mVerticesDrawn; }
+    GLsizeiptr getPrimitivesDrawn() const;
 
   private:
     friend class TransformFeedback;
@@ -93,13 +95,15 @@
     const OffsetBindingPointer<Buffer> &getIndexedBuffer(size_t index) const;
     size_t getIndexedBufferCount() const;
 
+    GLsizeiptr getVerticesDrawn() const { return mState.getVerticesDrawn(); }
+    GLsizeiptr getPrimitivesDrawn() const { return mState.getPrimitivesDrawn(); }
+
     // Returns true if any buffer bound to this object is also bound to another target.
     bool buffersBoundForOtherUse() const;
 
     angle::Result detachBuffer(const Context *context, GLuint bufferName);
 
-    rx::TransformFeedbackImpl *getImplementation();
-    const rx::TransformFeedbackImpl *getImplementation() const;
+    rx::TransformFeedbackImpl *getImplementation() const;
 
     void onBindingChanged(const Context *context, bool bound);
 